core/src/main/java/com/google/instrumentation/common/TimestampFactory.java

```diff
@@ -41,16 +41,18 @@ public static Timestamp now() {
   // This class cannot be final because in case of java8 runtime we can have a better granularity
   // timestamp.
   @Immutable
-  static class Handler {
-    protected Handler() {};
+  static interface Handler {
+    Timestamp timeNow();
+  }
 
-    protected Timestamp timeNow() {
+  static final class DefaultHandler implements Handler {
+    public Timestamp timeNow() {
```

Comment:
I think this should have `@Override`, though I'm surprised that CI didn't catch it.